### ISSUE TITLE
Acid Island - Inventory Duplication on Death

### ISSUE THREAD

**Issue by IvanStajdi** [tastybento/ASkyBlock-Bugs-N-Features#437]
Hello. Ever since I updated my server to Spigot 1.12 and re-added Acid Island (and Greenhouses), my players have been experiencing a strange issue in the Acid Island world. Whenever a player dies in the Acid Island world due to any cause and respawn, their inventory is duplicated. I have run several tests on this issue and have come up with a model of how it seems to work:

1. A player dies due to something on Acid Island, and since the gamerule keepInventory is false their items drop to the ground as they should. 
2. The player respawns back on their Acid Island, except their inventory isn't empty like it should be (since keepInventory is false). Instead their inventory is instantly filled with the items they had shortly before dying. 
3. The player then sucks in the items that dropped on the ground when they died due to keepInventory being false. Since they collected the dropped items in addition to the inventory that restored itself automatically, their inventory is now effectively duplicated.

Here is some info that might help. I am using the most recent update of Acid Island and Greenhouses. I am using the Multiverse plugin to link many worlds on my server, such as Survival, Creative, Acid Island and others. I also use the latest version of PerWorldInventory to separate inventories between worlds so players cannot, for an example, carry over a stack of diamond blocks from Creative to Survival. This glitch does not occur on any other world except for Acid Island. 

Another interesting fact is that the inventory that is restored automatically is not always exactly what the player had in their inventory when they died, it can also be something that they had a few minutes ago, which suggests to me that the inventory is "saved" every so often and then for some reason when a player dies the inventory is automatically restored to them from that save. Also, sometimes instead of the inventory being duplicated, the old inventory is simply restored and there are no dropped items to be collected so there is no duplication. However players can exploit this by placing all their items in a chest and then dying immedietaly afterwards in order to respawn with duplicates e of the items they placed in the chest, and they can do this possibly indefinitely. In this case, armor spawns in without any durability loss and already equipped on the player.

In order to fix the problem I have tried to remove the Acid Island worlds from PerWorldInventory's worlds.yml file and then reload the plugin and restart the server, which did nothing to fix the problem, so I re-added them. I made sure both Minecraft’s and Multiverse’s keepInventory gamerules were set to false. Then I tried setting the gamerule keepInventory to true, which simply made both inventories automatically appear instead of a copy being automatically restored and then the actual item drops are collected. I also looked for PerWorldInventory updates but it seems I have the most updated version. If you’re wondering why I didn’t try to replace PerWorldInventory with MultiInv it’s because I’ve had MultiInv on my server before and it started acting up after some time so I replaced it with PerWorldInventory which has worked fine since. I didn’t want to risk messing up the inventories of all the worlds in the process of trying to fix Acid Island’s inventory bug by re-adding MultiInv.

Then I attempted a fresh re-install of Acid Island’s latest version several times (meaning I deleted the Acid Island and Greenhouses plugins and their folders, deleted the Acid Island worlds and removed them from Multiverse and PerWorldInventories, then restarted the server, stopped it again, added Acid Island and Greenhouses and restarted again) with both adding it to PerWorldInventories’s worlds.yml and not, but nothing fixed the issue.  My server hosting’s customer support was also none the wiser.

Acid Island is one of mine as well as my playerbase’s favorite gamemodes and I think it is a brilliantly done plugin, so this troubles me greatly. I urgently await your response, as until this is solved my Acid Island world is effectively unplayable since even keeping inventory on death takes away from the intended challenge, and inventory duplication on death makes it broken and not challenging all, as I’m sure you understand. 

Thank you in advance.
Sincerely yours, IvanStajdi, owner of Ladybug MC.

**Comment by ResqDiver1317** [comment]
We had issues with this as well. We found that Multiverse's handling of the respawns was not playing nice with Essentials. Found a post on PWI's pages about it and someone said to disable bed spawning in the multiverse worlds file for all worlds. We did that on ours and it solved the issue. Now they are respawning back at the SkyBlock spawn and they have to go back to their islands, but the dup issue is resolved.

**Comment by tastybento** [comment]
AcidIsland saves a player's inventory when they die only if they are a visitor on another island and the setting **allowvisitorkeepinvondeath: true** is in the config. This is done at the point of death and when their items are saved, their inventory is completely cleared so nothing can drop. If **allowvisitorkeepinvondeath** is set to false, then AcidIsland isn't doing anything at all with inventories and death/respawn. In that case, the likelihood that it is due to another plugin is high. What is your setting? @ResqDiver1317's may be right. It's a good idea to try that.

**Comment by IvanStajdi** [comment]
Thank you ResqDiver1317 and tastybento. I checked my Acid Island config, allowvisitorkeepinvondeath was set to false all along. Then I changed the bedRespawn setting for all the worlds in MultiverseCore's worlds.yml file to 'false' and restarted my server, but unfortunately the glitch is still occuring as before. 

Also lately it seems that inventory duplication is rarer or doesn't happen, but the old inventory which the player had when they logged out last is returned on death. This is easily explotaible by logging out with valuables such as dirt blocks or tokens in inventory, logging back in and putting them in a chest, and then dying an indefinite amount of times to gain an indefinite amount of duplicates of those valuables. So until that is fixed my Acid Island is virtually unplayable.

**Comment by tastybento** [comment]
Inventories don't just magically appear so it must be another plugin. What other plugins do you have?

**Comment by IvanStajdi** [comment]
I went through my list of plugins and I don't think it's a problem with any of those plugins, I've had all those plugins since long ago and none of them really have anything to do with inventories. The only plugins I can see being related are Multiverse and PerWorldInventories, and Acid Island itself. The new plugins I've added to my server lately are AntiClone (prevents cloning maps), ArmorStandTools (self-explanatory), TokenManager (self-explanatory) and EssentialsGeoIP (to see what country new players are from so staff can greet them in the proper language). I don't see how any of those plugins would mess with inventories, and especially not only Acid Island's inventory.
